opendatahub-io/odh-dashboard
# Correctly render divider between sections on project details page

Close #1032 

## Description


Instead of adding the Divider by providing a prop named `includeDivider` in `EmptyDetailsList` component, I moved the divider to the render of the page. In each section, I added a prop named `isEmpty` to check whether the section is empty or not based on the context provider, then only render the divider below the section when the section is empty and it's not the last section on the page.

### Screenshots:
1. Before:
<img width="1904" alt="Screenshot 2023-03-27 at 2 27 06 PM" src="https://user-images.githubusercontent.com/37624318/227889597-f7e72fac-11e3-4aed-a0a6-bab4fa8bb679.png">

2. After the change, when all the sections are empty:
<img width="1904" alt="Screenshot 2023-03-27 at 4 26 06 PM" src="https://user-images.githubusercontent.com/37624318/227890438-f1485dc5-df1c-4ec6-b9f5-ebf050d1c803.png">

3. After the change, when the model serving feature is disabled (so the data connection is the last section):
<img width="1904" alt="Screenshot 2023-03-27 at 4 32 42 PM" src="https://user-images.githubusercontent.com/37624318/227890564-2366589d-22d9-4607-8202-0c7387b7b158.png">

4. After the change, when some sections are not empty:
<img width="1904" alt="Screenshot 2023-03-27 at 4 33 36 PM" src="https://user-images.githubusercontent.com/37624318/227890606-ed3d6ac9-d84f-4491-b76f-0a895b2e513c.png">
<img width="1904" alt="Screenshot 2023-03-27 at 4 34 08 PM" src="https://user-images.githubusercontent.com/37624318/227890689-919e4fd3-a214-4a98-bf5e-2ef5227bddc9.png">

Visual changes cc @kywalker-rh

## How Has This Been Tested?



1. Create a project, and check the dividers between the sections
2. Try to create and delete workbenches/storages/data connections... see if there are dividers. The correct result should be there are no dividers between sections when the top section is not empty and there is no divider at the bottom in any situation.
3. Try to disable the model serving feature by changing the dashboard config so the data connections section is the last one, and make sure you cannot see the divider under it no matter if there are data connections in the section or not.

## Test Impact


Add storybook test for project details page, check no dividers on default page (with data) and ${section number minus one} dividers on an empty page.

## Request review criteria:
